server/leafnode.go

```diff
@@ -2298,11 +2293,14 @@ func (c *client) processLeafSub(argo []byte) (err error) {
 	}
 
 	// If we are a hub check that we can publish to this subject.
-	if checkPerms && subjectIsLiteral(string(sub.subject)) && !c.pubAllowedFullCheck(string(sub.subject), true, true) {
-		c.mu.Unlock()
-		c.leafSubPermViolation(sub.subject)
-		c.Debugf(fmt.Sprintf("Permissions Violation for Subscription to %q", sub.subject))
-		return nil
+	if checkPerms {
+		subj := bytesToString(sub.subject)
```

Comment:
For `subjectIsLiteral()` it would be fine, but not for `c.pubAllowedFullCheck()` since again we will use `sl.Match()` in there.